js/app/articleHTMLRenderer.js

```diff
@@ -153,13 +153,28 @@ var ArticleHTMLRenderer = new Knowledge.Class({
         if (!model.original_uri || model.original_uri === '')
             return '';
 
+        function get_svg (uri) {
+            let file = Gio.file_new_for_uri(uri);
+
+            try {
+                let [success, svg] = file.load_contents(null);
+                return (success) ? svg.toString() : null;
```

Comment:
I think this still needs a small rework - `success` can never be false in these GIO functions since otherwise an exception is thrown. So it's better to always ignore it. (It's an annoying [bug](https://gitlab.gnome.org/GNOME/gjs/issues/68) in GJS but can't be changed unless we do a big API break.)

Here's what I would do:
```js
try {
    let [, svg] = file.load_contents(null);
    return svg.toString();
} catch (e) {
    return null;
}
```

Reply:
ohh right, in my narrow C api view I did not realized that it will always return true :)